planner/core/rule_join_reorder.go

```diff
@@ -78,18 +79,23 @@ func (s *joinReOrderSolver) optimizeRecursive(ctx sessionctx.Context, p LogicalP
 				baseSingleGroupJoinOrderSolver: baseGroupSolver,
 				eqEdges:                        eqEdges,
 			}
-			p, err = groupSolver.solve(curJoinGroup)
+			newPlan, err = groupSolver.solve(curJoinGroup)
 		} else {
 			dpSolver := &joinReorderDPSolver{
 				baseSingleGroupJoinOrderSolver: baseGroupSolver,
 			}
 			dpSolver.newJoin = dpSolver.newJoinWithEdges
-			p, err = dpSolver.solve(curJoinGroup, expression.ScalarFuncs2Exprs(eqEdges))
+			newPlan, err = dpSolver.solve(curJoinGroup, expression.ScalarFuncs2Exprs(eqEdges))
 		}
 		if err != nil {
 			return nil, err
 		}
-		return p, nil
+		newJoin, ok1 := newPlan.(*LogicalJoin)
+		oldJoin, ok2 := p.(*LogicalJoin)
+		if ok1 && ok2 {
+			newJoin.cartesianJoin = oldJoin.cartesianJoin
```

Comment:
please add a test case for this.

Reply:
this is for sql like `select * from t t1, t t2 where t1.a = t2.a and t2.a = 1`.
after `ppdSolver`, it becomes `select * from t t1, t t2 where t1.a = 1 and t2.a = 1`, so that after `joinReOrderSolver`, original cartesian information lost. 

There are already so many testcase for `line 82`, no need to add more testcase.

For 88, a new testcase will be add to `explain_test.go/TestExplainCartesianJoin` by set session variable `tidb_opt_join_reorder_threshold`